manager/src/main/java/com/redhat/service/smartevents/manager/models/CloudProvider.java

```diff
@@ -72,4 +74,12 @@ public List<CloudRegion> getRegions() {
     public void setRegions(List<CloudRegion> regions) {
         this.regions = regions;
     }
+
+    @JsonIgnore
+    public Optional<CloudRegion> getRegionByName(String name) {
+        if (this.regions == null) {
+            return Optional.empty();
+        }
+        return this.regions.stream().filter(r -> r.getName().equals(name)).findFirst();
```

Comment:
Could we please use `Objects.equals(r.getName(), name)` instead of ` r.getName().equals(name)`? The first mentioned should be safe from `NullPointerException`.